sqs/queue_template.go

```diff
@@ -0,0 +1,168 @@
+	// message is delivered from the queue. This blocks
+	// other components from receiving the same message
+	// and gives the initial component time to process
+	// and delete the message from the queue.
+	// Values must be from 0 to 43,200 seconds (12 hours). If you don't specify a value, AWS CloudFormation uses the default value of 30 seconds.
+	VisibilityTimeout int `json:"visibility_timeout,omitempty"`
+	// Tags are the AWS resource tags to attach to this queue.
+	Tags map[string]string `json:"-"`
+}
+
+// GetStackTemplate returns a cloudformation Template for provisioning an SQS queue
+func QueueTemplate(params QueueParams) (*goformation.Template, error) {
+	template := goformation.NewTemplate()
+
+	tags := []goformationtags.Tag{}
+	for k, v := range params.Tags {
+		tags = append(tags, goformationtags.Tag{
+			Key:   k,
+			Value: v,
+		})
+	}
+
+	var redrivePolicy interface{}
+	if params.RedriveMaxReceiveCount > 0 {
+		redrivePolicy = map[string]interface{}{
+			"deadLetterTargetArn": goformation.GetAtt(SQSDLQResourceName, "Arn"),
+			"maxReceiveCount":     params.RedriveMaxReceiveCount,
+		}
+	} else {
+		redrivePolicy = ""
+	}
+
+	template.Resources[SQSResourceName] = &goformationsqs.Queue{
+		QueueName: params.QueueName,
+		Tags: append(tags, goformationtags.Tag{
+			Key:   "QueueType",
+			Value: "Main",
+		}),
+		ContentBasedDeduplication:     params.ContentBasedDeduplication,
+		DelaySeconds:                  params.DelaySeconds,
+		FifoQueue:                     params.FifoQueue,
+		MaximumMessageSize:            params.MaximumMessageSize,
+		MessageRetentionPeriod:        params.MessageRetentionPeriod,
+		ReceiveMessageWaitTimeSeconds: params.ReceiveMessageWaitTimeSeconds,
+		RedrivePolicy:                 redrivePolicy,
+		VisibilityTimeout:             params.VisibilityTimeout,
+	}
+
+	dlQueueName := fmt.Sprintf("%s-dl", params.QueueName)
+	template.Resources[SQSDLQResourceName] = &goformationsqs.Queue{
+		QueueName: dlQueueName,
+		Tags: append(tags, goformationtags.Tag{
+			Key:   "QueueType",
+			Value: "Dead-Letter",
+		}),
+		FifoQueue:                 params.FifoQueue,
+		MessageRetentionPeriod:    params.MessageRetentionPeriod,
+		ContentBasedDeduplication: params.ContentBasedDeduplication,
+		VisibilityTimeout:         params.VisibilityTimeout,
+	}
+
+	template.Outputs[SQSQueueURLOutputName] = goformation.Output{
+		Description: "SQSQueue URL",
+		Value:       goformation.Ref(SQSResourceName),
+		Export: goformation.Export{
+			Name: fmt.Sprintf("%s-%s", params.QueueName, SQSQueueURLOutputName),
+		},
+	}
+
+	template.Outputs[SQSQueueARNOutputName] = goformation.Output{
+		Description: "SQSQueue ARN",
+		Value:       goformation.GetAtt(SQSResourceName, "Arn"),
+		Export: goformation.Export{
+			Name: fmt.Sprintf("%s-%s", params.QueueName, SQSQueueARNOutputName),
+		},
+	}
+
+	template.Outputs[SQSDLQueueURLOutputName] = goformation.Output{
+		Description: "SQSQueue DLQ URL",
```

Comment:
for consistency with the below:
```suggestion
		Description: "SQSDLQueue URL",
```

Reply:
Did something similar by adding "main" vs "deadletter" ... no need for the crazy golang naming style in descriptions for humans